src/queires/useGetComments.ts

```diff
@@ -0,0 +1,29 @@
+import { useInfiniteQuery } from '@tanstack/react-query';
+
+import { Comment, getComments } from '@src/apis/comment';
+import { IGetCommentsResponseData } from '@src/apis/comment';
+
+import { queryKeys } from './constant';
+
+export const useGetComments = (topicId: number) => {
+  const result = useInfiniteQuery<IGetCommentsResponseData>(
+    [queryKeys.comment, topicId],
+    ({ pageParam }) => getComments(topicId, pageParam),
+    {
+      getNextPageParam: (lastPage) => lastPage.offsetId ?? undefined,
+    },
+  );
+
+  const { data } = result;
+
+  const comments = data
+    ? data.pages
+        .map((page) => page.data)
+        .reduce((mergedContents: Comment[], currentContents) => [...mergedContents, ...(currentContents || [])], [])
+    : [];
+
+  return {
+    ...result,
+    data: comments,
+  };
```

Comment:
```suggestion
  return {
    ...result,
    comments,
  };
```
data로 내리지 말고 comments로 내려도 괜찮지 않을까요?

Reply:
음.. 이 부분은 다른 곳과 일관성있게 처리학 싶어서... `data` 로 내리게 하려고 했는데 각 도메인에 맞는 형태로 내리는게 더 좋을까요??

Reply:
개인적으로는 매번 쓰는 곳에서 `data: comment` 같은 형식으로 또 쓰게되니
어차피 그럴바에는 그냥 그 상태로 주는 게 낫지 않나 싶기도 하다가
쓰다보니 결국 refetch같은 경우도 애매해지긴 하네요.... 🤔

Reply:
음.. 저도 이 부분 좀 고민되네요ㅠㅠ